File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(const std::vector<int>& prices) {
        if (prices.empty()) return 0; // Handle edge case where prices vector might be empty

        int minPrice = prices[0];
        int maxProfit = 0;

        for (int price : prices) {
            if (price < minPrice) {
                minPrice = price;  // Update the minimum price found so far
            } else if (price - minPrice > maxProfit) {
                maxProfit = price - minPrice;  // Update the maximum profit if current profit is higher
            }
        }
        return maxProfit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min_price = float('inf')
        max_profit = 0
        for num in prices:
            if num < min_price:
                min_price = num
            if (num - min_price) > max_profit:
                max_profit = num - min_price
        

        return max_profit